src/Data/HashTable/ST/Basic.hs

```diff
@@ -620,6 +681,12 @@ recordIsDeleted = (== deletedMarker)
 
 
 ------------------------------------------------------------------------------
+{-# INLINE recordIsFilled #-}
+recordIsFilled :: Elem -> Bool
+recordIsFilled el = not (recordIsEmpty el) && not (recordIsDeleted el)
```

Comment:
It's not that those bindings aren't there in older GHC, it's that the type changed from 

```haskell
==# :: Int# -> Int# -> Bool
```

to

```haskell
==# :: Int# -> Int# -> Int#
```

This change happened in GHC 7.8. The former version is not desirable since the return type of "Bool" creates a branch that can't be eliminated, but the new one is branch free (a "`sub`" followed by a `setz`).